flink-connector-mysql-cdc/src/main/java/com/alibaba/ververica/cdc/connectors/mysql/source/reader/MySqlSourceReader.java

```diff
@@ -145,31 +148,33 @@ public void handleSourceEvents(SourceEvent sourceEvent) {
             LOG.info(
                     "The subtask {} receive request to report finished snapshot splits.",
                     subtaskId);
-            reportFinishedSnapshotSplits(finishedNoAckSplits.values());
+            reportFinishedSnapshotSplits();
             // also try to request new split
             context.sendSplitRequest();
         } else {
             super.handleSourceEvents(sourceEvent);
         }
     }
 
-    private void reportFinishedSnapshotSplits(Collection<MySqlSplit> splits) {
-        if (!splits.isEmpty()) {
-            final ArrayList<Tuple2<String, BinlogOffset>> finishedNoAckSplits = new ArrayList<>();
-            for (MySqlSplit split : splits) {
-                finishedNoAckSplits.add(Tuple2.of(split.getSplitId(), split.getHighWatermark()));
+    private void reportFinishedSnapshotSplits() {
+        if (!finishedNoAckSplits.isEmpty()) {
+            final ArrayList<Tuple2<String, BinlogOffset>> noAckSplits = new ArrayList<>();
+            for (MySqlSplit split : finishedNoAckSplits.values()) {
+                if (split.isSnapshotSplit()) {
+                    noAckSplits.add(
```

Comment:
Can we force to add `SnapshotSplit` in `finishedNoAckSplits`? That would be much simpler.